swarm/src/lib.rs

```diff
@@ -603,38 +607,56 @@ where
                 Poll::Pending => network_not_ready = true,
                 Poll::Ready(NetworkEvent::ConnectionEvent { connection, event }) => {
                     let peer = connection.peer_id();
-                    let connection = connection.id();
-                    this.behaviour.inject_event(peer, connection, event);
+                    let conn_id = connection.id();
+                    if !this.banned_peer_connections.contains(&conn_id)
+                        || this.banned_peers.contains(&peer)
+                    {
+                        this.behaviour.inject_event(peer, conn_id, event);
+                    } else {
+                        log::debug!(
+                            "Ignoring event from disallowed connection or peer: {:?} {:?}.",
+                            peer,
+                            conn_id,
+                        );
+                    }
                 }
                 Poll::Ready(NetworkEvent::AddressChange {
                     connection,
                     new_endpoint,
                     old_endpoint,
                 }) => {
                     let peer = connection.peer_id();
-                    let connection = connection.id();
-                    this.behaviour.inject_address_change(
-                        &peer,
-                        &connection,
-                        &old_endpoint,
-                        &new_endpoint,
-                    );
+                    let conn_id = connection.id();
+                    if !this.banned_peer_connections.contains(&conn_id) {
+                        this.behaviour.inject_address_change(
+                            &peer,
+                            &conn_id,
+                            &old_endpoint,
+                            &new_endpoint,
+                        );
+                    }
                 }
                 Poll::Ready(NetworkEvent::ConnectionEstablished {
                     connection,
-                    num_established,
+                    established_ids,
                     concurrent_dial_errors,
                 }) => {
                     let peer_id = connection.peer_id();
                     let endpoint = connection.endpoint().clone();
                     if this.banned_peers.contains(&peer_id) {
+                        // Mark the connection for the banned peer as disallowed.
```

Comment:
```suggestion
                        // Mark the connection for the banned peer as banned, thus withholding any future events from the connection to the behaviour.
```

This might seem like a nit pick, but still I wonder whether we need to differentiate in _banned_ peers and _disallowed_ connections. Why not call them _banned_ connections instead of _disallowed_ connections? As a newcomer I would find this more intuitive.